File: chess_cpp/piecies/Pawn.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "Pawn.h"

void Pawn::setMoved(bool moved) {
    this->moved = moved;
}

std::vector<Spot> Pawn::getPathSpots(Board *board, Spot *start, Spot *end) {
    std::vector<Spot> spots;
    if (end->getY() - start->getY() == 2)
        spots.push_back(board->getBox(start->getX(), start->getY() + 1));

    return spots;
}

void Pawn::move(Board *board, Spot *start, Spot *end) {
    // Each piece is responsible for its logic of movement.
    // So the movement logic should be implemented here.
    // the method is not implemented, as the requirement for current task is only validation
}

bool Pawn::isEnPassant() {
    return enPassant;
}

void Pawn::setEnPassant(bool enPassant) {
    this->enPassant = enPassant;
}

bool Pawn::isMoved() {
    return moved;
}

File: oop_chess/src/chess/piecies/Pawn.java
package chess.piecies;

import chess.Board;
import chess.Spot;

import java.util.ArrayList;
import java.util.List;

public class Pawn extends Piece{
    private boolean enPassant;
    private boolean isMoved;

    public Pawn(boolean white) {
        super(white);
    }

    @Override
    public List<Spot> getPathSpots(Board board, Spot start, Spot end) {
        List<Spot> spots = new ArrayList<>();
        if(end.getY() - start.getY() == 2)
            spots.add(board.getBox(start.getX(), start.getY() + 1));

        return spots;
    }

    @Override
    public void move(Board board, Spot start, Spot end) {
        // Each piece is responsible for its logic of movement.
        // So the movement logic should be implemented here.
        // the method is not implemented, as the requirement for current task is only validation
    }

    public boolean isEnPassant() {
        return enPassant;
    }

    public void setEnPassant(boolean enPassant) {
        this.enPassant = enPassant;
    }

    public boolean isMoved() {
        return isMoved;
    }

    public void setMoved(boolean moved) {
        isMoved = moved;
    }
}
